tools/dotnet-linker/Steps/GenerateReferencesStep.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Mono.Cecil;
+
+using Xamarin.Bundler;
+using Xamarin.Linker;
+
+namespace Xamarin {
+
+	public class GenerateReferencesStep : ConfigurationAwareStep {
+		protected override string Name { get; } = "Generate References";
+		protected override int ErrorCode { get; } = 2320;
+
+		protected override void TryEndProcess ()
+		{
+			base.TryEndProcess ();
+
+			var app = Configuration.Application;
+			var required_symbols = Configuration.DerivedLinkContext.RequiredSymbols;
+			var items = new List<MSBuildItem> ();
+				
+			switch (app.SymbolMode) {
+			case SymbolMode.Ignore:
+				break;
+			case SymbolMode.Code:
+				string reference_m = Path.Combine (Configuration.CacheDirectory, "reference.m");
+				reference_m = Configuration.Target.GenerateReferencingSource (reference_m, required_symbols);
+				if (!string.IsNullOrEmpty (reference_m)) {
+					var item = new MSBuildItem { Include = reference_m };
+					items.Add (item);
+				}
+				Configuration.WriteOutputForMSBuild ("_ReferencesFile", items);
+				break;
+			case SymbolMode.Linker:
+			case SymbolMode.Default:
```

Comment:
The default mode should already be changed to one of the other modes:

https://github.com/xamarin/xamarin-macios/blob/725190243995c63d26c89c2a200f1cea07fb82b6/tools/common/Application.cs#L673-L679

so the `Default` case can be removed here